server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -927,6 +997,11 @@ protected void onFoundPeersUpdated() {
 
     private void startElectionScheduler() {
         assert electionScheduler == null : electionScheduler;
+
+        if (getLocalNode().isMasterNode() == false) {
```

Comment:
can you make a separate PR with this change (+ corresponding assertions + test). This issue has caused build failures (see https://elasticsearch-ci.elastic.co/view/All/job/elastic+elasticsearch+zen2+feature-branch-periodic/16/testReport/junit/org.elasticsearch.client.transport/NodeDisconnectIT/testNotifyOnDisconnect/ ).

Reply:
Yes, I opened #35996.